src/AltinnCore/Runtime/RestControllers/ProcessController.cs

```diff
@@ -41,29 +66,445 @@ public class ProcessController : ControllerBase
+                return Conflict($"Process is not started. Use start!");
+            }
+
+            ProcessKeeper = BpmnReader.Create(processService.GetProcessDefinition(org, app));
+
+            string currentElementId = instance.Process.CurrentTask?.ElementId;
+
+            if (currentElementId == null)
+            {
+                return Conflict($"Instance does not have current task information!");
+            }            
+
+            string nextElement = CheckOrGetValidNextElement(currentElementId, elementId, out ActionResult nextElementError);
+            if (nextElementError != null)
+            {
+                return nextElementError;
+            }
+
+            if (InstanceIsValid(instance))
+            {
+                Instance changedInstance = await UpdateProcessStateToNextElement(org, app, instance, instanceOwnerId, instanceGuid, nextElement);
+
+                return Ok(changedInstance.Process);
+            }
+            else
+            {
+                return Conflict("Cannot complete/close current task {currentTaskId}. Task is not valid!");
+            }
+        }
+
+        /// <summary>
+        /// Attemts to close the process by running start and next until an end event is reached.
+        /// </summary>
+        /// <returns></returns>
+        [HttpPut("completeProcess")]
+        public async Task<ActionResult<ProcessState>> CompleteProcess(
             [FromRoute] string org,
             [FromRoute] string app,
             [FromRoute] int instanceOwnerId,
             [FromRoute] Guid instanceGuid)
         {
             Instance instance = await instanceService.GetInstance(app, org, instanceOwnerId, instanceGuid);
-            string currentTaskId = instance.Process?.CurrentTask?.ProcessElementId;
 
-            return Ok();           
+            if (instance == null)
+            {
+                return NotFound("Cannot find instance");
+            }
+
+            ProcessKeeper = BpmnReader.Create(processService.GetProcessDefinition(org, app));
+
+            if (instance.Process == null)
+            {
+                string startEvent = GetValidStartEvenOrError(null, out ActionResult startEventError);
+
+                if (startEventError != null)
+                {
+                    return startEventError;
+                }
+
+                instance = await StartProcessOfInstance(org, app, instanceOwnerId, instanceGuid, instance, startEvent);                
+            }
+
+            string currentTaskId = instance.Process.CurrentTask?.ElementId ?? instance.Process.StartEvent;
+
+            if (currentTaskId == null)
+            {
+                return Conflict($"Instance does not have valid currentTask");
+            }
+       
+            // do next until end event is reached or task cannot be completed.
+            int counter = 0;
+            do
+            {
+                List<string> nextElements = ProcessKeeper.NextElements(currentTaskId);
```

Comment:
you chould check if the instance is valid at this point to save many unnecessary operations if it is in an invalid state.